File: backend/src/main/java/com/xxgwy/common/response/R.java
package com.xxgwy.common.response;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class R<T> {
    private int code;
    private String message;
    private T data;

    public static <T> R<T> ok(T data) {
        return new R<>(200, "ok", data);
    }

    public static <T> R<T> ok() {
        return new R<>(200, "ok", null);
    }

    public static <T> R<T> ok(String message, T data) {
        return new R<>(200, message, data);
    }

    public static <T> R<T> failed(int code, String message) {
        return new R<>(code, message, null);
    }

    public static <T> R<T> failed(String message) {
        return new R<>(500, message, null);
    }
}

File: backend/src/main/kotlin/com/xxgwy/common/response/R.kt
package com.xxgwy.common.response

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class R<T>(
    val code: Int = 200,
    val message: String = "ok",
    val data: T? = null
) {
    companion object {
        @JvmStatic
        fun <T> ok(): R<T> = R()

        @JvmStatic
        fun <T> ok(data: T): R<T> = R(data = data)

        @JvmStatic
        fun <T> failed(message: String): R<T> = R(500, message)

        @JvmStatic
        fun <T> failed(code: Int, message: String): R<T> = R(code, message)
    }
}
